name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/viewers/TransactionOwnerListEditingSupport.java

```diff
@@ -0,0 +1,182 @@
+package name.abuchen.portfolio.ui.util.viewers;
+
+import java.beans.PropertyDescriptor;
+import java.util.ArrayList;
+import java.util.List;
+
+import name.abuchen.portfolio.model.Account;
+import name.abuchen.portfolio.model.AccountTransaction;
+import name.abuchen.portfolio.model.AccountTransferEntry;
+import name.abuchen.portfolio.model.BuySellEntry;
+import name.abuchen.portfolio.model.Client;
+import name.abuchen.portfolio.model.Portfolio;
+import name.abuchen.portfolio.model.PortfolioTransaction;
+import name.abuchen.portfolio.model.PortfolioTransferEntry;
+import name.abuchen.portfolio.model.Transaction;
+import name.abuchen.portfolio.model.TransactionOwner;
+import name.abuchen.portfolio.model.TransactionPair;
+
+/**
+ * Creates a cell editor with a combo box of Accounts. Options must be a list of non-null
+ * values. By default, the user must choose one option. Override the method
+ * {@link #canBeNull} in order to add an additional empty element
+ */
+public class TransactionOwnerListEditingSupport extends ListEditingSupport
+{
+    private Class<?> subjectType;
+    private String attributeName;
+    private Client client;
+
+    public TransactionOwnerListEditingSupport(Client client, Class<?> subjectType, String attributeName)
+    {
+        super(subjectType, attributeName, new ArrayList<Object>());
+        this.subjectType = subjectType;
+        this.attributeName = attributeName;
+        this.client = client;
+    }
+
+    private Transaction getTransaction(Object element)
+    {
+        Transaction t;
+        if (element instanceof Transaction)
+            t = (Transaction) element;
+        else if (element instanceof TransactionPair)
+            t = (Transaction) ((TransactionPair) element).getTransaction();
+        else
+            throw new UnsupportedOperationException();
+        return t;
+    }
+
+    @Override
+    public boolean canEdit(Object element)
+    {
+        Transaction t = getTransaction(element);
+
+        if (t.getCrossEntry() == null)
+            return false;
+
+        boolean canEdit = (adapt(t.getCrossEntry()) != null); 
+        if (canEdit)
+        {
+            List<?> options; 
+            if (attributeName.equals("transactionOwner"))
+            {
+                if (t.getCrossEntry().getOwner(t) instanceof Account)
+                    options = client.getAccounts();
```

Comment:
Die Kontobuchungen können nicht einfach in eine anderes Konto übernommen werden. Die Währung jeder Kontobuchung muss mit der Währung des Kontos übereinstimmen. Ansonsten müssen noch Forex Units mit dem Wechselkurs angelegt werden.

Der Einfachheit halber schlage ich vor in der Drop-Down Liste nur Konten anzuzeigen, die die gleiche Währung haben. Wenn man eine USD Buchung auf ein EUR Konto umzieht, muss man eben die Dialog öffnen und die Wechselkurse eingeben.

Reply:
Hallo @buchen,
valider Hinweis hier unterschiedliche Währungen zu beachten.
Mein Problem war hier, die Liste der Accounts zu editieren, da diese "unmodifieable" ist.
Hast Du da einen Tip?
Wollte auf diese Art auch verhinden, daß man Buchungen auf sich selber macht. Die werden zwar ignoriert, sind aber zunächst wählbar...

Danke.
cmaoling

Reply:
> Mein Problem war hier, die Liste der Accounts zu editieren, da diese "unmodifieable" ist.
Hast Du da einen Tip?

Ich denke Du musst eine Kopie der Liste machen nur mit den Konten der passenden Währung a la ```getAccounts().stream().filter(a -> currency.equals(a.getCurrencyCode)).collectTo(Collectors.toList())```

Reply:
e21c5173 filtert nun die Liste, das Accounts mit anderer Währung nichtmehr angezeigt werden.